huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_data_standard.go

```diff
@@ -0,0 +1,582 @@
+	curArray := curJson.([]interface{})
+	for _, v := range curArray {
+		id := utils.PathSearch("id", v, "").(string)
+		fdName := utils.PathSearch("fd_name", v, "").(string)
+		rst[fdName] = id
+	}
+	return rst
+}
+
+func buildUpdateDataStandardBodyParams(d *schema.ResourceData, dataStandardNameToIdMap map[string]string) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"id":           d.Id(),
+		"directory_id": d.Get("directory_id"),
+		"values":       buildUpdateDataStandardRequestBodyValue(d, dataStandardNameToIdMap),
+	}
+	return bodyParams
+}
+
+func buildUpdateDataStandardRequestBodyValue(d *schema.ResourceData, dataStandardNameToIdMap map[string]string) []map[string]interface{} {
+	rawArray := d.Get("values").(*schema.Set).List()
+	if len(rawArray) == 0 {
+		return nil
+	}
+
+	rst := make([]map[string]interface{}, len(rawArray))
+	for i, v := range rawArray {
+		raw := v.(map[string]interface{})
+		rst[i] = map[string]interface{}{
+			"id":           dataStandardNameToIdMap[raw["fd_name"].(string)],
+			"fd_name":      raw["fd_name"],
+			"fd_value":     raw["fd_value"],
+			"directory_id": d.Get("directory_id"),
+		}
+	}
+	return rst
+}
+
+func resourceDataStandardDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// deleteDataStandard: delete DataArtsStudio data standard
+	var (
+		deleteDataStandardHttpUrl = "v2/{project_id}/design/standards"
+		deleteDataStandardProduct = "dataarts"
+	)
+	deleteDataStandardClient, err := cfg.NewServiceClient(deleteDataStandardProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio client: %s", err)
+	}
+
+	deleteDataStandardPath := deleteDataStandardClient.Endpoint + deleteDataStandardHttpUrl
+	deleteDataStandardPath = strings.ReplaceAll(deleteDataStandardPath, "{project_id}", deleteDataStandardClient.ProjectID)
+	deleteDataStandardPath = strings.ReplaceAll(deleteDataStandardPath, "{id}", d.Id())
+
+	deleteDataStandardOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders: map[string]string{
+			"workspace": d.Get("workspace_id").(string),
+		},
+	}
+
+	deleteDataStandardOpt.JSONBody = utils.RemoveNil(buildDeleteDataStandardBodyParams(d))
+	_, err = deleteDataStandardClient.Request("DELETE", deleteDataStandardPath, &deleteDataStandardOpt)
+	if err != nil {
+		return diag.Errorf("error deleting DataArts Studio data standard: %s", err)
+	}
+
+	return nil
+}
+
+func buildDeleteDataStandardBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"ids": []string{d.Id()},
+	}
+	return bodyParams
+}
+
+// resourceDataStandardImportState use to import an id with format <workspace_id>/<id>
+func resourceDataStandardImportState(_ context.Context, d *schema.ResourceData, _ interface{}) ([]*schema.ResourceData, error) {
+	if !strings.Contains(d.Id(), "/") {
+		return []*schema.ResourceData{d}, nil
+	}
```

Comment:
There is unnecessary to check whether id string contains character `/`

Reply:
done